lib/index.js

```diff
@@ -469,6 +471,74 @@ class AuthenticationPlugin {
     return this.getUsersRepository().delete(user._id, {refresh: 'wait_for'});
   }
 
+  /**
+   * Search users by ES query concerning exclusively their username
+   *
+   * @param {Object} searchBody - Search request body (query, sort, etc.)
+   * @param {Object} options - from (undefined), size (undefined)
+   *
+   * @returns {Promise.<{ hits, total }>}
+   */
+  async search (searchBody, { from, size } = {}) {
+    debug('exists(searchBody = %j)', searchBody);
+
+    let sort;
+    let query;
+    let isRequiredFieldsInBody = false;
+    const forbiddenFields = extractESMappingFields(storageMapping.users);
+    const forbiddenKeywords = [
+      'multi_match',
+    ]; // @todo Keep up to date this blacklist to prevent queries on forbiddenFields
+
+    const sanitizeSearchBody = (key, value) => {
+      if (key === 'username') {
+        isRequiredFieldsInBody = true;
+        return { key: '_id' };
+      }
+      else if (value === 'username') {
+        isRequiredFieldsInBody = true;
+        return { value: '_id' };
+      }
+      else if (forbiddenKeywords.includes(key)) {
+        throw new this.context.errors.BadRequestError(`The "${key}" keyword is not allowed in this search query for security concerns.`);
+      }
+      else if (forbiddenFields.some((property) => property === key || property === value)) {
+        throw new this.context.errors.ForbiddenError(`Forbidden field "${key}". Only the "username" or "kuid" fields are sortable and only the first is also searchable.`);
+      }
+      return {};
+    };
+
+    if (isPlainObject(searchBody.query)) {
+      query = objectRecursiveMap(searchBody.query, sanitizeSearchBody);
+      if (! isRequiredFieldsInBody) {
+        throw new this.context.errors.ForbiddenError('Only the "username" field is searchable, otherwise leave the query empty.');
```

Comment:
`ForbiddenError` are used when you try to access a resource you are not authorized to.
IMHO  a `BadRequestError` is more appropriate here

Reply:
This case and the [following](https://github.com/kuzzleio/kuzzle-plugin-auth-passport-local/pull/76#discussion_r637425758) are trickier: we don't know exactly, either there is no username in the request or there is one inside an array with probably other forbidden fields / wildcards. Therfore, I think your `BadRequestError` suggestion must apply because we are unable to find the forbidden fields/wildcard (if there's any) which would led to a `ForbiddenError`.

Reply:
Agreed :+1: